lib/IR/ChildMemSizeBasedScheduler.cpp

```diff
@@ -81,103 +81,130 @@ void ChildMemSizeBasedScheduler::computeNodeComputationMaxMemorySize() {
 /// Order children by (maxSize - resultSize). It gives more
 /// priority to the nodes that free more memory after
 /// their computation.
-void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *N) {
-  // Each child should be scheduled just once.
-  if (isScheduled(N))
-    return;
+void ChildMemSizeBasedScheduler::orderChildNodesAndSchedule(Node *node) {
   // Do not explicitly schedule storage nodes.
-  if (isa<Storage>(N))
+  if (isa<Storage>(node)) {
     return;
-  // A set of node's sorted children.
-  llvm::SmallVector<Node *, 8> orderedChildren;
-  for (int idx = 0, e = N->getNumInputs(); idx < e; ++idx) {
-    orderedChildren.push_back(N->getNthInput(idx));
   }
-
-  if (N->hasPredicate()) {
-    orderedChildren.push_back(N->getPredicate());
+  // Each child should be scheduled just once.
+  if (isScheduled(node)) {
+    return;
   }
 
-  // We don't model memory dependencies, but we still need to honor them.
-  // Make sure the a node mutating any of its inputs happens after the last
-  // non-mutating use of the operand being mutated. Some examples of such nodes
-  // would be SaveNode and QuantizationProfileNode.
-  for (unsigned idx = 0, e = N->getNumInputs(); idx < e; ++idx) {
-    // We don't care about inputs that are not mutated by the node.
-    if (!N->isOverwrittenNthInput(idx)) {
+  std::vector<Node *> orderedChildren;
```

Comment:
Can you add a comment for this variable? Is seems like you use it as a worklist or work-queue, right?

Also, is the name `orderedChildren` still the right name given that you use it for a broader purpose now?

Reply:
Renamed to `worklist` and added a description about usage of `worklist` and `readyNodes`.